streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -83,7 +84,12 @@ import org.apache.spark.storage.StorageLevel
  * }}}
  */
 @DeveloperApi
-abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable {
+abstract class Receiver[T](val storageLevel: StorageLevel)(implicit t: ClassTag[T])
```

Comment:
It is really trickier to add ClassTag in many place to support **automatically-pick-best-serializer feature**.